src/main/java/uk/gov/hmcts/reform/managecase/api/payload/CaseAssignmentRequest.java

```diff
@@ -21,7 +21,7 @@
     private String caseTypeId;
 
     @JsonProperty("case_id")
-    @NotEmpty(message = "Case ID can not be empty")
+    @NotEmpty(message = "Case ID has to be a valid 16-digit Luhn number")
```

Comment:
'Not empty' and invalid seem like different tests to me.  Also there are ways to validate the luhn number, see [ACA-51 /api/payload/RequestedCaseUnassignment.java#L26-L28](https://github.com/hmcts/aac-manage-case-assignment/blob/ACA-51/src/main/java/uk/gov/hmcts/reform/managecase/api/payload/RequestedCaseUnassignment.java#L26-L28).

Also it may be worth copying the ACA-51 file [api/errorhandling/ValidationError.java](https://github.com/hmcts/aac-manage-case-assignment/blob/ACA-51/src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/ValidationError.java).  and replacing all reference to the validation messages in your models, swagger [(see example)](https://github.com/hmcts/aac-manage-case-assignment/blob/ACA-51/src/main/java/uk/gov/hmcts/reform/managecase/api/controller/CaseAssignmentController.java#L168-L190) and unit tests.  **NB: our ACA-51 code hasn't been reviewed yet but I feel it is much cleaner with all strings defined once.**

Reply:
I think `luhn number` validation should be part of CCD as it owns the data / format.   MCA just need to validate non empty value .

Reply:
Better to keep in MCA, actually. Doing proactive data validation has its own pro's and con's and it was a deliberate design approach taken to do it in MCA for its pro's overweighing con's. MCA will suffer from performance issues already, due to it's nature to call API's of multiple parties, and anything that it can do in its local with reasonable cost, it should do. Lunh number convention for case id's is a very well-settled domain rule which is safe to re-implement at every frontier line.

Reply:
I wouldn't call CCD for the validation purpose but it'll be delegated (covered in ) to CCD as part of Case search by ID.   I didn't see any benefit of replicating same in MCA; instead additional maintenance responsibility. What about assignee idam ID validation then?

Reply:
Didn't mean you'd call data store for validation. But the requests to it due to malformed case id's would take far longer to conclude with a negative response from mca, than in the case of a front-line validation. Luhn number convention on case ids is a rather stable rule, which means the maintenance of it will not be any significant effort. On the other hand, if we rely on downstream's validation, then we will have the code to selectively and correctly reflect such validation failures to upstream, which means another piece (albeit small) of error handling code to be maintained. If we had a similar, stable validation rule to my knowledge about IDAM user IDs I'd surely include it in the LLD by the same token. But IDAM IDs don't have such a common convention. I recall in the database old records of users having very different structure of IDAM IDs.

Reply:
FYI: marking this conversation as resolved to allow progress into QA review and as I also agree that adding a Luhn check is a safe and useful addition to MCA APIs whilst an equivalent IDAM ID validation is not possible.